lib/rucio/core/meta.py

```diff
@@ -63,7 +64,14 @@ def add_key(key, key_type, value_type=None, value_regexp=None, session=None):
     try:
         new_key.save(session=session)
     except IntegrityError as error:
-        if error.args[0] == "(IntegrityError) column key is not unique":
+        if error.args[0] == "(IntegrityError) column key is not unique" \
+           or ('UNIQUE constraint failed' in error.args[0]) \
+           or ('conflicts with persistent instance' in error.args[0]) \
+           or match('.*IntegrityError.*ORA-00001: unique constraint.*DID_KEYS_PK.*violated.*', error.args[0]) \
+           or match('.*IntegrityError.*1062.*Duplicate entry.*for key.*', error.args[0]) \
+           or match('.*IntegrityError.*duplicate key value violates unique constraint.*', error.args[0]) \
+           or match('.*UniqueViolation.*duplicate key value violates unique constraint.*', error.args[0]) \
+           or match('.*sqlite3.IntegrityError.*is not unique.*', error.args[0]):
```

Comment:
Is there no way to simplify this? It is quite clumsy and could cause issues if any of these error messages change in the future. Have you asked Martin about this?

Reply:
I haven't spoken to Martin; I based this off what I saw other core functions doing (e.g. `add_dids()`). All the ones I saw use this kind of logic when excepting an `IntegrityError` and raising a `Duplicate`. Some places don't have the if clause and just raise it as a generic `RucioException`, but that won't be caught by the bootstrap process and so the second test run will fail. If there is a better way of doing it I'm happy to change it, but from looking at the code it doesn't like there is (or at least we don't do it).

Reply:
Ah, if there is precedent elsewhere in the code, we can leave this as it is